### PR TITLE
Correct typo in Sensor: GCSObjectsWtihPrefixExistenceSensor 

### LINKED ISSUE #14097: Typo in Sensor: GCSObjectsWtihPrefixExistenceSensor (should be GCSObjectsWithPrefixExistenceSensor)
Typo in Google Cloud Storage sensor: airflow/providers/google/cloud/sensors/gcs/GCSObjectsWithPrefixExistenceSensor

The word _With_ is spelled incorrectly. It should be: GCSObjects**With**PrefixExistenceSensor

**Apache Airflow version**: 2.0.0
**Environment**:
- **Cloud provider or hardware configuration**: Google Cloud
- **OS** (e.g. from /etc/os-release): Mac OS BigSur

### PR DESCRIPTION
**Pull request by morrme**
This PR fixes the issue described in [Typo in Sensor: GCSObjectsWtihPrefixExistenceSensor (should be GCSObjectsWithPrefixExistenceSensor)](https://github.com/apache/airflow/issues/14097)

Closes #14097

### CODE REVIEW

**Review by potiuk — APPROVED** [review]
Nice!

**Review by turbaszek — APPROVED** [review]

**Comment by potiuk** [comment]
Unfortunately tests fail :(

**Comment by morrme** [comment]
> Unfortunately tests fail :(
 @potiuk 

Yes 💔 ...  this is related to the changes made in the `deprecated_classes` file. I am still looking into what the correct change should have been.

File: tests/always/test_project_structure.py [hunk 1]
```diff
@@ -220,7 +220,7 @@ class TestGoogleProviderProjectStructure(unittest.TestCase):
         'airflow.providers.google.cloud.operators.compute.ComputeEngineBaseOperator',
         'airflow.providers.google.cloud.sensors.gcs.GCSObjectExistenceSensor',
         'airflow.providers.google.cloud.sensors.gcs.GCSObjectUpdateSensor',
-        'airflow.providers.google.cloud.sensors.gcs.GCSObjectsWtihPrefixExistenceSensor',
+        'airflow.providers.google.cloud.sensors.gcs.GCSObjectsWithPrefixExistenceSensor',
```
Comment (turbaszek) [inline tests/always/test_project_structure.py:223]:
```suggestion
        'airflow.providers.google.cloud.sensors.gcs.GCSObjectsWithPrefixExistenceSensor',
        'airflow.providers.google.cloud.sensors.gcs.GCSObjectsWtihPrefixExistenceSensor',
```

**Review by turbaszek — APPROVED** [review]

**Pull request merged**